bokehjs/src/lib/models/glyphs/varea_step.ts

```diff
@@ -0,0 +1,177 @@
+import {PointGeometry} from "core/geometry"
+import {FloatArray, ScreenArray} from "core/types"
+import {Area, AreaView, AreaData} from "./area"
+import {Context2d} from "core/util/canvas"
+import {SpatialIndex} from "core/util/spatial"
+import * as hittest from "core/hittest"
+import * as p from "core/properties"
+import {StepMode} from "core/enums"
+import {Selection} from "../selections/selection"
+
+export type VAreaStepData = AreaData & {
+  _x: FloatArray
+  _y1: FloatArray
+  _y2: FloatArray
+
+  sx: ScreenArray
+  sy1: ScreenArray
+  sy2: ScreenArray
+}
+
+export interface VAreaStepView extends VAreaStepData {}
+
+export class VAreaStepView extends AreaView {
+  declare model: VAreaStep
+  declare visuals: VAreaStep.Visuals
+
+  protected _index_data(index: SpatialIndex): void {
+    const {min, max} = Math
+
+    for (let i = 0; i < this.data_size; i++) {
+      const x = this._x[i]
+      const y1 = this._y1[i]
+      const y2 = this._y2[i]
+      index.add_rect(x, min(y1, y2), x, max(y1, y2))
+    }
+  }
+
+  protected _step_path(ctx: Context2d, mode: StepMode, sx: ScreenArray, sy: ScreenArray, from_i: number, to_i: number): void {
+    // Assume the path was already moved to the first point
+    let prev_x = sx[from_i]
+    let prev_y = sy[from_i]
+    const idx_dir = from_i < to_i ? 1 : -1
+    for (let i = from_i + idx_dir; i != to_i; i += idx_dir) {
+      switch (mode) {
+        case "before":
+          ctx.lineTo(prev_x, sy[i])
+          ctx.lineTo(sx[i], sy[i])
+          break
+        case "after":
+          ctx.lineTo(sx[i], prev_y)
+          ctx.lineTo(sx[i], sy[i])
+          break
+        case "center":
+          const mid_x = (prev_x + sx[i]) / 2
+          ctx.lineTo(mid_x, prev_y)
+          ctx.lineTo(mid_x, sy[i])
+          ctx.lineTo(sx[i], sy[i])
+          break
+      }
+      prev_x = sx[i]
+      prev_y = sy[i]
+    }
+  }
+
+  protected _flip_mode(mode: StepMode): StepMode {
+    switch (mode) {
+      case "before":
+        return "after"
+      case "after":
+        return "before"
+      case "center":
+        return "center"
+    }
+  }
```

Comment:
`_flip_mode` is duplicated in both files, maybe it  could be a public free function `flip_step_mode` in utils, etc?

Reply:
Good idea! I've done that and also applied your other findings, thanks.